### PR TITLE
Avoid installing `init.d` services when `init.d` is completely deprecated

### PR DESCRIPTION
**Pull request by verdx**
|Related issue|
|---|
| #1918 |

## Description

This PR removes all errors derived from the complete deprecation of `init.d` in some systems, such as Red Hat Enterprise Linux 9 or systems derived from it. 

- The deprecation is detected by checking the existence of file `/usr/lib/systemd/systemd-sysv-install`.

- On the case of the Wazuh agent and manager, service file `/etc/rc.d/init.d/wazuh-[agent|manager]` is not installed in the first place.

- On the case of the Wazuh indexer, as the file is installed by the Open Search package, it is now removed when necessary.

## Logs example

All logs before and after the changes can be seen in the issue.

## Tests
All tests are explained in the issue.

### CODE REVIEW

File: rpms/SPECS/wazuh-agent.spec [hunk 1]
```diff
@@ -35,6 +35,7 @@ log analysis, file integrity monitoring, intrusions detection and policy and com
 ./gen_ossec.sh conf agent centos %rhel %{_localstatedir} > etc/ossec-agent.conf
 
 %build
+%define initd_valid %( if [ -f /usr/lib/systemd/systemd-sysv-install ]; then echo "1" ; else echo "0"; fi )
```
Comment (DFolchA) [inline rpms/SPECS/wazuh-agent.spec:38]: According to documentation:
```
%define has local scope, which means that it applies only to a specified part of a SPEC file. In addition, the body of a %define macro is expanded when used—​it is lazily evaluated
```
So this macro will only affect the `%build` section.

Also, the `%build` section is only executed during package generation and not during the installation so the value of this macro will be taken from the machine where the package was built.

File: rpms/SPECS/wazuh-manager.spec [hunk 2]
```diff
@@ -36,6 +36,7 @@ log analysis, file integrity monitoring, intrusions detection and policy and com
 ./gen_ossec.sh conf manager centos %rhel %{_localstatedir} > etc/ossec-server.conf
 
 %build
+%define initd_valid %( if [ -f /usr/lib/systemd/systemd-sysv-install ]; then echo "1" ; else echo "0"; fi )
```
Comment (DFolchA) [inline rpms/SPECS/wazuh-manager.spec:39]: Same.

**Review by DFolchA — CHANGES_REQUESTED** [review]

**Comment by verdx** [comment]
# Update

### Previous code
After @DFolchA's commentary and tests on `init.d`-using systems, such as `SLES 11` and `CentOS 6`, it has been proved all changes simply removed the `init.d` service, no matter the system. The macro used, `%define` was replaced during the package building and not the package install and so the service was never installed.

### Current code architecture
Right now, the idea for a solution on this matter is different on the manager, indexer and agent:
- On the manager and indexer, as all supported systems use `systemd`, the `init.d` service has been completely removed. This way, if any new system completely deprecates `init.d`, as `Red Hat Linux 9` and its derivatives have done, no problem will arise.
- The main idea for the agent is to remove the `init.d` service on the post installation of the package if the system uses `systemd`. The main problem right now has come in identifying if `systemd` is being used in a correct way for all possible systems. 

### Identifying if `systemd` is being used
There are a few possible ways to try and see if `systemd` is being used
- `ps --no-headers -p 1 -o comm=` returns "systemd" if it is being used
- According to the [documentation](https://www.freedesktop.org/software/systemd/man/sd_booted.html), the way the own `systemctl` checks for it is by doing this check `[[ -d /run/systemd/system ]]`, whether directory `/run/systemd/system` exists.
- Other option is checking if command `systemctl` is present, but it is not a very clean way, and the command can be removed. 

The main problem is docker, as this methods work on normal hosts, but docker containers do not really use `systemd`, so the real accurate methods don't really work, which is a problem when doing tests. 


## New GitHub Action tests
To test the different options that affect this PR, new tests have been added, mainly adding new systems to the install and enable test and also trying to test the init.d systems too, although it isn't clear this tests can be done or are really useful. They are all tests done on Docker, which, for `init` tests, is not at all the best platform.

File: stack/indexer/rpm/build_package.sh [hunk 3]
```diff
@@ -158,6 +156,8 @@ main() {
         esac
     done
 
+    set -ex
```
Comment (DFolchA) [inline stack/indexer/rpm/build_package.sh:159]: Remove this.

**Review by DFolchA — CHANGES_REQUESTED** [review]
The restart of the services after the upgrade should take this change into consideration.

File: stack/indexer/rpm/build_package.sh (hunk 3 above)
Comment (verdx) [reply stack/indexer/rpm/build_package.sh:159]: This debugging tool is necessary for the Jenkins builds

**Review by DFolchA — APPROVED** [review]
GJ.

**Review by davidcr01 — APPROVED** [review]
Nice work.

**Pull request merged**